libs/application_runner/lib/environment/query_server.ex

```diff
@@ -64,8 +64,9 @@ defmodule ApplicationRunner.Environment.QueryServer do
          {:ok, coll} <- Keyword.fetch(opts, :coll),
          {:ok, query_transformed} <- Keyword.fetch(opts, :query_transformed),
          {:ok, query_parsed} <- Keyword.fetch(opts, :query_parsed),
-         {:ok, data} <- fetch_initial_data(env_id, coll, query_transformed),
-         {:ok, projection} <- Keyword.fetch(opts, :projection) do
+         {:ok, projection} <- Keyword.fetch(opts, :projection),
+         {:ok, options} <- Keyword.fetch(opts, :options),
+         {:ok, data} <- fetch_initial_data(env_id, coll, query_transformed, projection, options) do
```

Comment:
The initial data should not be fetched with the projections and options since they both should be applied after.
The initial data must stay full to apply the projections and options.

Reply:
Ok so this PR will need a lot more work to handle each mongo aggregators by hand. 

We will start by just implementing the "skip" and "limit" aggregators and we will add more in the future.

Reply:
Yes. You can create an Epic with the list of the aggregators to manage as we did for the operators